### PR TITLE
Audit/issue2

### PR DESCRIPTION
**Pull request by stanieltron**
addressed potential integer underflow with replacing sub with checked_sub

### CODE REVIEW

**Review by iStrike7 — APPROVED** [review]

File: pallets/xyk/src/lib.rs [hunk 1]
```diff
@@ -1614,8 +1615,10 @@ impl<T: Config> XykFunctionsTrait<T::AccountId> for Pallet<T> {
 				sold_asset_id.into(),
 				&sender,
 				&vault,
-				(sold_asset_amount - buy_and_burn_amount - treasury_amount - pool_fee_amount)
-					.into(),
+				(sold_asset_amount
+					.checked_sub(buy_and_burn_amount + treasury_amount + pool_fee_amount)
```
Comment (mateuszaaa) [inline pallets/xyk/src/lib.rs:1619]: what about possible overflow with `+`. Shouldt we use checked_add everywhere ?

File: pallets/xyk/src/lib.rs [hunk 2]
```diff
@@ -1784,8 +1787,10 @@ impl<T: Config> XykFunctionsTrait<T::AccountId> for Pallet<T> {
...
```
Comment (mateuszaaa) [inline pallets/xyk/src/lib.rs:1791]: checked add ?

**Review by mateuszaaa — APPROVED** [review]
overall LGTM ,but few comments just to be sure

File: pallets/xyk/src/lib.rs (hunk 1 above)
Comment (stanieltron) [reply pallets/xyk/src/lib.rs:1619]: buy_and_burn_amount and the rest are 0.05% of sold amount which is capped u128, so no need

**Pull request merged**